### ISSUE TITLE
some question about train with dataset NPS

### ISSUE THREAD

**Issue by Yipzcc** [tusharsangam/TransVisDrone#10]
Dear author, thank you very much for your contributions. I have a question that I would like to ask you.

when i train NPS with your model or other model like yolov5s, yolov5l, yolov5l-xs-tph, I always get unsatisfactory results, with very low AP values, as shown in the figure below.
![2023-11-22 08-12-20 的屏幕截图](https://github.com/tusharsangam/TransVisDrone/assets/33560875/71cd9f40-4b8c-4978-b20e-b9cdf4d08848)

Here are some of the training commands.

1.yolov5l move out the self.temporaltransformers in yolo.py
python train.py --img 1280 --adam --batch-size 8 --epochs 80 --data ./data/NPS_1.yaml --weights ./weights/yolov5l.pt   \
--hy runs_ori/train/NPS/image_size_1280_temporal_YOLO5l_5_frames_NPS_end_to_end_skip_0/hyp_skip0_frames1.yaml \
--cfg models/yolov5l.yaml --name 1280_1gpu_b8_oriv5l_pre_1 --project ./runs/train/NPS --single-cls --device 0

2.yolov5l-xs-tph    （move out the self.temporaltransformers in yolo.p
python train.py --img 1280 --adam --batch-size 4 --epochs 80 --data ./data/NPS_1.yaml --weights ./weights/yolov5l.pt   \
--hy runs_ori/train/NPS/image_size_1280_temporal_YOLO5l_5_frames_NPS_end_to_end_skip_0/hyp_skip0_frames1.yaml \
--cfg models/yolov5l-xs-tph.yaml --name 1280_1gpu_b8_oriv5ltph_pre_1 --project ./runs/train/NPS --single-cls

i have checked out my dataset labels and found the  the distribution of label data is almost identical to what you have uploaded, as shown in the figure below. I believe there is no problem with the data. Could you please give me some suggestions on how to solve this issue?
![labels](https://github.com/tusharsangam/TransVisDrone/assets/33560875/8393826e-18ae-4bea-a49f-d9fc7f9babee)

**Comment by defrag-bambino** [comment]
Hello my friend,
where can I download NPS and FLDrones datasets? 

Thanks

**Comment by tusharsangam** [comment]
@defrag-bambino Have you tried running the pretrained model first ?
NPS dataset can be downloaded from https://engineering.purdue.edu/~bouman/UAV_Dataset/
for annotations please follow https://github.com/mwaseema/Drone-Detection these are different than original annotations, please use these.
FL-drones can be obtained from their respective authors, I got it from my co-author or I can send over email

**Comment by tusharsangam** [comment]
I will be uploading a detailed README soon.

**Comment by defrag-bambino** [comment]
Thanks!
A detailed README with instructions on how to train etc. would be ideal. Because ultimately I want to train on a custom dataset.

**Comment by zhuang5252** [comment]
> 亲爱的作者，非常感谢您的贡献。我有一个问题想问你。
> 
> 当我使用您的模型或其他模型（如 yolov5s、yolov5l、yolov5l-xs-tph）训练 NPS 时，我总是得到不令人满意的结果，AP 值非常低，如下图所示。 ![2023-11-22 08-12-20 的屏幕截图](https://private-user-images.githubusercontent.com/33560875/284751532-71cd9f40-4b8c-4978-b20e-b9cdf4d08848.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU4MjU4ODQsIm5iZiI6MTcxNTgyNTU4NCwicGF0aCI6Ii8zMzU2MDg3NS8yODQ3NTE1MzItNzFjZDlmNDAtNGI4Yy00OTc4LWIyMGUtYjljZGY0ZDA4ODQ4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE2VDAyMTMwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA1N2M4NTM2YjdjNzc5N2YyZDc3N2U1NzZjMzk2MjMwNWZlMTYwOTdmMjkzMTcyYjFiMzdlNjRiYjMyZTE1ZDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fz3xGr2a4EoTTPqTyC-9R1u4Oq0gBuh8ig0mmRElN_Q)
> 
> 以下是一些训练命令。
> 
> 1.yolov5l（移出 yolo.py 中的self.temporaltransformers） python train.py --img 1280 --adam --batch-size 8 --epochs 80 --data ./data/NPS_1.yaml --weights ./weights/yolov5l.pt --hy runs_ori/train/NPS/image_size_1280_temporal_YOLO5l_5_frames_NPS_end_to_end_skip_0/hyp_skip0_frames1.yaml --cfg models/yolov5l.yaml --name 1280_1gpu_b8_oriv5l_pre_1 --project ./runs/train/NPS --single-cls --device 0
> 
> 2.yolov5l-xs-tph （移出 yolo.py 中的 self.temporaltransformers） 蟒蛇 train.py --img 1280 --亚当 --batch-size 4 --epochs 80 --data ./data/NPS_1.yaml --weights ./weights/yolov5l.pt --hy runs_ori/train/NPS/image_size_1280_temporal_YOLO5l_5_frames_NPS_end_to_end_skip_0/hyp_skip0_frames1.yaml --cfg models/yolov5l-xs-tph.yaml --name 1280_1gpu_b8_oriv5ltph_pre_1 --project ./runs/train/NPS --单cls
> 
> 我检查了我的数据集标签，发现标签数据的分布与您上传的数据几乎相同，如下图所示。我相信数据没有问题。你能给我一些关于如何解决这个问题的建议吗？ ![标签](https://private-user-images.githubusercontent.com/33560875/284755057-8393826e-18ae-4bea-a49f-d9fc7f9babee.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU4MjU4ODQsIm5iZiI6MTcxNTgyNTU4NCwicGF0aCI6Ii8zMzU2MDg3NS8yODQ3NTUwNTctODM5MzgyNmUtMThhZS00YmVhLWE0OWYtZDlmYzdmOWJhYmVlLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE2VDAyMTMwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWExZTZmMTUzZGNhNDAxMGVlMDZkNTZkZjk3MmU0N2ZiMzViMzk3M2U2NGJjYTRkZDNhMDY1NTBlOTJjZDFiMTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.h0HcZX8HoZSw1ZTlNRCYKhohAqmBcDeqsEGPECwe7tA)

Hello, I also encountered this issue when using YOLOv5n, YOLOv5s, YOLOv5l, and YOLOv5l xs tph. My accuracy, recall, and map did not increase but decreased. Have you resolved this issue?
![微信图片_20240516102429](https://github.com/tusharsangam/TransVisDrone/assets/78992752/3970cec1-9eac-49d5-a829-19a5874f5c11)
![微信图片_20240516102445](https://github.com/tusharsangam/TransVisDrone/assets/78992752/6fa40019-4c17-4756-9ec7-e69ba770eca0)

**Comment by Yipzcc** [comment]
github太慢了，我csdn博客烤荔枝的大猫，这个代码nps数据集序号和dataset里面不一致，dataset对nps数据集会➕1
---- 回复的原邮件 ----
发件人 ***@***.***> 日期 2024年05月16日 10:26 收件人 ***@***.***> 抄送至 ***@***.***>***@***.***> 主题 Re: [tusharsangam/TransVisDrone] some question about train with dataset NPS (Issue #10) 
亲爱的作者，非常感谢您的贡献。我有一个问题想问你。
当我使用您的模型或其他模型（如 yolov5s、yolov5l、yolov5l-xs-tph）训练 NPS 时，我总是得到不令人满意的结果，AP 值非常低，如下图所示。 
以下是一些训练命令。
1.yolov5l（移出 yolo.py 中的self.temporaltransformers） python train.py --img 1280 --adam --batch-size 8 --epochs 80 --data ./data/NPS_1.yaml --weights ./weights/yolov5l.pt --hy runs_ori/train/NPS/image_size_1280_temporal_YOLO5l_5_frames_NPS_end_to_end_skip_0/hyp_skip0_frames1.yaml --cfg models/yolov5l.yaml --name 1280_1gpu_b8_oriv5l_pre_1 --project ./runs/train/NPS --single-cls --device 0
2.yolov5l-xs-tph （移出 yolo.py 中的 self.temporaltransformers） 蟒蛇 train.py --img 1280 --亚当 --batch-size 4 --epochs 80 --data ./data/NPS_1.yaml --weights ./weights/yolov5l.pt --hy runs_ori/train/NPS/image_size_1280_temporal_YOLO5l_5_frames_NPS_end_to_end_skip_0/hyp_skip0_frames1.yaml --cfg models/yolov5l-xs-tph.yaml --name 1280_1gpu_b8_oriv5ltph_pre_1 --project ./runs/train/NPS --单cls
我检查了我的数据集标签，发现标签数据的分布与您上传的数据几乎相同，如下图所示。我相信数据没有问题。你能给我一些关于如何解决这个问题的建议吗？ 
Hello, I also encountered this issue when using YOLOv5n, YOLOv5s, YOLOv5l, and YOLOv5l xs tph. My accuracy, recall, and map did not increase but decreased. Have you resolved this issue?
_20240516102429.png (view on web)
_20240516102445.png (view on web)

**Comment by zhuang5252** [comment]
> @defrag-bambino Have you tried running the pretrained model first ? NPS dataset can be downloaded from https://engineering.purdue.edu/~bouman/UAV_Dataset/ for annotations please follow https://github.com/mwaseema/Drone-Detection these are different than original annotations, please use these. FL-drones can be obtained from their respective authors, I got it from my co-author or I can send over email

Hello, I would like to use the following FL Dornes dataset. Could you please help me contact them?

**Comment by tusharsangam** [comment]
@defrag-bambino  try to reach out to any of these three authors; You should be able to find their emails from the given paper citation.
A. Rozantsev, V. Lepetit, and P. Fua, “Detecting flying objects using
a single moving camera,” IEEE transactions on pattern analysis and
machine intelligence, vol. 39, no. 5, pp. 879–892, 2016.

**Comment by zhuang5252** [comment]
> 尝试联系这三位作者中的任何一位;您应该能够从给定的论文引用中找到他们的电子邮件。 A. Rozantsev、V. Lepetit 和 P. Fua，“使用单个移动相机检测飞行物体”，IEEE Transactions on Pattern Analysis and Machine Intelligence，第 39 卷，第 5 期，第 879–892 页，2016 年。

Okey, thank you very much.

**Comment by wish44165** [comment]
> github太慢了，我csdn博客烤荔枝的大猫，这个代码nps数据集序号和dataset里面不一致，dataset对nps数据集会➕1
> ---- 回复的原邮件 ----
> 发件人 ***@***.***> 日期 2024年05月16日 10:26 收件人 ***@***.***> 抄送至 ***@***.***>***@***.***> 主题 Re: [tusharsangam/TransVisDrone] some question about train with dataset NPS (Issue [#10](https://github.com/tusharsangam/TransVisDrone/issues/10))
> 亲爱的作者，非常感谢您的贡献。我有一个问题想问你。
> 当我使用您的模型或其他模型（如 yolov5s、yolov5l、yolov5l-xs-tph）训练 NPS 时，我总是得到不令人满意的结果，AP 值非常低，如下图所示。
> 以下是一些训练命令。
> 1.yolov5l（移出 yolo.py 中的self.temporaltransformers） python train.py --img 1280 --adam --batch-size 8 --epochs 80 --data ./data/NPS_1.yaml --weights ./weights/yolov5l.pt --hy runs_ori/train/NPS/image_size_1280_temporal_YOLO5l_5_frames_NPS_end_to_end_skip_0/hyp_skip0_frames1.yaml --cfg models/yolov5l.yaml --name 1280_1gpu_b8_oriv5l_pre_1 --project ./runs/train/NPS --single-cls --device 0
> 2.yolov5l-xs-tph （移出 yolo.py 中的 self.temporaltransformers） 蟒蛇 train.py --img 1280 --亚当 --batch-size 4 --epochs 80 --data ./data/NPS_1.yaml --weights ./weights/yolov5l.pt --hy runs_ori/train/NPS/image_size_1280_temporal_YOLO5l_5_frames_NPS_end_to_end_skip_0/hyp_skip0_frames1.yaml --cfg models/yolov5l-xs-tph.yaml --name 1280_1gpu_b8_oriv5ltph_pre_1 --project ./runs/train/NPS --单cls
> 我检查了我的数据集标签，发现标签数据的分布与您上传的数据几乎相同，如下图所示。我相信数据没有问题。你能给我一些关于如何解决这个问题的建议吗？
> Hello, I also encountered this issue when using YOLOv5n, YOLOv5s, YOLOv5l, and YOLOv5l xs tph. My accuracy, recall, and map did not increase but decreased. Have you resolved this issue?
> _20240516102429.png (view on web)
> _20240516102445.png (view on web)
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the thread.Message ID: ***@***.***>

Hi @Yipzcc @tusharsangam @DAVEISHAN ,

I've also encountered the same issue, the AP is lower than expected.

I checked the processed data, and it seems correct.

Have you figured out the cause of the problem and how to solve it?

I look forward to your response. Thank you for your assistance.

**Comment by tusharsangam** [comment]
@wish44165 @Yipzcc 
Yes I have recently reproduced nps results 
You can follow thsese steps for NPS dataset, there might be an annotation mismatch when preprocessing nps dataset on main branch. I fixed them in nps_reproduce branch.
Please follow these steps:
for NPS dataset, switch the git branch to 'nps_pretrained_reproduce', go to folder reproduce, 
Step 1. Run videos to frames extraction using videos_to_frames_nps.py (modify input, outptut paths)
Step 2. Then run nps_to_yolo.py to convert nps format to yolo format & take care of the above index offset.
Step 3. Verify the annotation to frame mapping by plotting few videos with plot_annotation_on_video.py
Afterwards, use the same branch to train the nps model using appropriate commands in 'submit-train.slurm' or verify the pretrained model on validation/test dataset by running commands in `submit-reproduce.slurm`
If this still doesn't work let me know & I can run this again on my end.
I recommend that you process these frames & annotations again from this branch.

**Comment by wish44165** [comment]
> use the same branch to train the nps model using appropriate commands in 'submit-train.slurm' or verify the pretrained model on validation/test dataset by running commands in `submit-reproduce.slurm`
> If this still doesn't work let me know & I can run this again on my end.
> I recommend that you process these frames & annotations

Hi @tusharsangam , thanks for your prompt response.

I'll test it this weekend and keep you updated on any progress, thank you.

**Comment by wish44165** [comment]
Hi @tusharsangam ,

I've trained the updated version, and it works well.

<img width="1493" height="434" alt="Image" src="https://github.com/user-attachments/assets/376e4e67-421d-40cf-939a-0a23ddc50823" />

I also ran inference with the trained model, and the visual performance looks great.

You've done an excellent job, this model seems highly suitable for video object tracking, especially for thermal infrared videos. Feel free to try it out and have fun in the [competition](https://www.codabench.org/competitions/9888/) 🤗

Thanks again for your great work!

**Comment by tusharsangam** [comment]
@wish44165  Awesome will update my main readme for NPS dataset & will close this issue. Thank you for confirming & I will take a look at the competition you mentioned.
